snap/local/start-mongod.sh

```diff
@@ -1,5 +1,14 @@
 #!/bin/bash
 
 # For security measures, daemons should not be run as sudo. Execute mongod as the non-sudo user: snap-daemon.
-exec $SNAP/usr/bin/setpriv --clear-groups --reuid snap_daemon \
-  --regid snap_daemon -- $SNAP/usr/bin/mongod --config ${SNAP_DATA}/etc/mongod/mongod.conf ${MONGOD_ARGS} "$@"
+exec \
+  "${SNAP}/usr/bin/setpriv" \
+  --clear-groups \
+  --reuid snap_daemon \
+  --regid snap_daemon \
+  -- \
+  "${SNAP}/usr/bin/mongod" \
+      --config "${SNAP_DATA}/etc/mongod/mongod.conf" \
+      --setParameter processUmask=037 \
```

Comment:
forgive my noob question but why do we allow `7: Everyone can read, write and execute`? My intuition says it should be `660`?

Reply:
So the way the mask works is the actual opposite of "adding permissions" - you should reason with the idea of "removing permissions", so a mask of `037` essentially means:
- do not remove any permission from the owner 
- remove  `wx` from the group
- remove all permissions `rwx` from others
Given that the log files are originally created with `600` permissions, the end resulted permissions after applying the said mask would be: `640`

Reply:
Ah thank you for explaining Mehdi